lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,970 @@
+			gren:Remove()
+			end
+		end)
+	end
+end
+
+
+if(SERVER)then
+KLEINER_BULLIES = KLEINER_BULLIES or {}
+end
+ 
+ 
+function ENT:PosInRange(pos)
+	local heightlimit = self.TargetHeight*1
+	local horizontalrange = (self:GetPos()*Vector(1,1,0)):Distance(pos*Vector(1,1,0))
+	local heightdiff = math.abs((self:GetPos() - pos).z)
+	if(horizontalrange < self.TargetHeightInnerRadius)then
+		heightlimit = self.TargetHeightInner
+	end
+	return horizontalrange <= self.LoseTargetDist and heightdiff <= heightlimit
+end
+ 
+function ENT:CanBeTarget(ent)
+	if(!IsValid(ent))then return false end
+	if(ent == self)then return false end
+
+	if(ent.InVehicle and ent:InVehicle())then return false end
+	if(ent.Alive and !ent:Alive())then return false end
+	if(ent:GetMoveType() == MOVETYPE_FLY)then return false end
+	if(player.IsAFK and player:IsAFK())then return false end
+	if(Safe and Safe(ent) and ent:IsPlayer())then return false end
+	if(self.TargetBlacklist and self.TargetBlacklist[ent] and self.TargetBlacklist[ent] > CurTime())then return false end
+	if(!self:PosInRange(ent:GetPos()))then return false end
+	return true
+end
+
+function ENT:CanBecomeTarget(ent) --use this if you want to add special requirements for the entity to become a target
+	if(self:GetRangeTo(ent) > self.SearchRadius and self:GetTargetPriority(ent) < 100 )then return false end
+	return self:CanBeTarget(ent)
+end
+
+function ENT:NearTarget()
+	if(!IsValid(self:GetTarget()))then return false end
+	return self:GetPos():Distance(self:GetTarget():GetPos()) < 90
+end
+
+
+function ENT:GetTargetPriority(ent)
+	if(!IsValid(ent))then return 0 end
+	local priority = ent:IsPlayer() and 1 or 0.05 --base amount
+	if(ent == KLEINER_OVERRIDE_TARGET)then priority = 100 end
+	if(self:GetTargetViolence(ent) > 0)then priority = priority * (1+(self:GetTargetViolence(ent)/10)) end -- 10% gain based on aggression towards kleiner
+	priority = priority * (1  + (math.Clamp(self.LoseTargetDist - self:GetRangeTo(ent),0,self.LoseTargetDist) / self.LoseTargetDist)/5) --up to 20% gain based on proximity
+	return priority
+end
+
+function ENT:GetTargetViolence(ent)
+	if(!IsValid(ent))then return -1 end
+	if(ent:GetClass() == "kleiner")then return -1 end
+	if(ent:IsPlayer() and KLEINER_BULLIES[ent:SteamID()])then return KLEINER_BULLIES[ent:SteamID()] end 
+	return 0
+end
+
+function ENT:SetTargetViolence(ent,amount)
+	if(!IsValid(ent))then return end
+	if(ent:IsPlayer())then 		
+			KLEINER_BULLIES[ent:SteamID()] = amount
+			--[[
+			net.Start("kleinernpc_warning")
+			net.WriteInt(amount,16)
+			net.Send(ent)
+			]]
+	end
+end
+
+function ENT:AddTargetViolence(ent)
+	if(!IsValid(ent))then return end
+	if(ent:IsPlayer())then self:SetTargetViolence(ent,self:GetTargetViolence(ent) + 1)  end
+end
+
+function ENT:HaveTarget()
+	local target = self:GetTarget()
+	if ( IsValid(target) and self:CanBeTarget(target) ) then
+		return true 
+	end
+	return self:FindTarget()
+end
+
+function ENT:GetTargetTable()
```

Comment:
Where is this used? Is there a reason you can't just use `KLEINER_NPC_TARGETS or {}` ?

Reply:
i think i was originally using this in place of KLEINER_NPC_TARGETS before i had that table. i will remove this